File: WebSinglePageAngularApp/src/app/_models/device.ts
import { User } from "./user";
import { Field } from "./field";
import { Action } from "./action";

export interface Device {
    id?: string;
    user?: User;
    userId?: string;
    connectionString?: string;
    name: string;
    online?: boolean;
    fields?: Array<Field>
    actions?: Array<Action>
    collaborators?: Array<User>
}

File: Microservices/DevicesService/DevicesService/Models/Device.cs
using IoTHubAPI.Models.Dtos;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace IoTHubAPI.Models
{
    public class Device
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        [Required]
        public string Id { get; set; }

        [Required]
        public string UserId { get; set; }

        [Required]
        public string ConnectionString { get; set; }

        [Required]
        public string Name { get; set; }
        public List<UserInfoDto> Collaborators { get; set; } = new List<UserInfoDto>();

    }
}
